meteor/client/ui/SegmentTimeline/Parts/SegmentTimelinePart.tsx

```diff
@@ -473,20 +505,36 @@ export class SegmentTimelinePartClass extends React.Component<Translated<WithTim
 		}
 	}
 
+	private getFutureShadePaddingTime = () => {
+		if (this.props.autoNextPart) {
+			return 0
+		}
+
+		/**
+		 * How far into the live part we are, in milliseconds.
+		 */
+		const timeIntoLivePart = Math.max(0, (this.props.livePosition || 0) - (this.props.livePartStartsAt || 0))
+
+		/**
+		 * The maximum amount of live line time padding to add, in milliseconds.
+		 */
+		const maxPadding = SegmentTimelinePartClass.getLiveLineTimePadding(this.props.timeScale)
+
+		/**
+		 * The amount of live line time padding to add, based on some simple math, in milliseconds.
+		 */
+		const computedPadding = Math.max(0, timeIntoLivePart + maxPadding - (this.props.livePartDisplayDuration || 0))
```

Comment:
Comments should explain `why` something is happening, not `what` is happening.

`timeIntroLivePart` could be renamed to `millisecondsIntoLivePart`.

`maxPadding` could be renamed to `maximumLiveLinePaddingInMilliseconds`.

`computedPadding` could be renamed to `liveLinePaddingToAddInMilliseconds`.

// All above names are just suggestions

The rule of thumb is: If it's difficult to find a good name for something (and hence need `what` comments), you should probably to something different.

Sidenote: It feels weird to talk about padding in milliseconds. Is padding the right word to use?